soh/soh/Enhancements/randomizer/location_access.h

```diff
@@ -12,18 +12,25 @@
 #define TIME_PASSES true
 #define TIME_DOESNT_PASS false
 
-typedef bool (*ConditionFn)();
+using ConditionFn = std::function<bool()>;
 
 // I hate this but every alternative I can think of right now is worse
 extern Rando::Context* ctx;
 extern std::shared_ptr<Rando::Logic> logic;
 
 class Region;
 
+constexpr std::string CleanCheckConditionString(std::string condition);
+
+#define EVENT_ACCESS(event, condition) \
+    EventAccess(                       \
```

Comment:
we're thinking of moving these to a bitset anyways so event would also be an enum